solr/core/src/test/org/apache/solr/search/mlt/CloudMLTQContentParserTest.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    Arrays.sort(actualIds);
+    Arrays.sort(expectedIds);
+    assertArrayEquals(expectedIds, actualIds);
+  }
+
+  @Test
+  public void testBoost() throws Exception {
+
+    QueryResponse queryResponse =
+        cluster
+            .getSolrClient()
+            .query(
+                COLLECTION,
+                new SolrQuery(
+                    "q",
+                    "{!mlt_content qf=lowerfilt_u boost=true mindf=0}" + seventeenth,
+                    "fq",
+                    "-id:17"));
+    SolrDocumentList solrDocuments = queryResponse.getResults();
+    int[] expectedIds = new int[] {7, 9, 13, 14, 15, 16, 20, 22, 24, 32};
+    int[] actualIds = new int[solrDocuments.size()];
+    int i = 0;
+    for (SolrDocument solrDocument : solrDocuments) {
+      actualIds[i++] = Integer.parseInt(String.valueOf(solrDocument.getFieldValue("id")));
+    }
+
+    Arrays.sort(actualIds);
+    Arrays.sort(expectedIds);
+    assertArrayEquals(expectedIds, actualIds);
+
+    String thirtineenth = "The quote red fox jumped over the lazy brown dogs." + "red green yellow";
+    queryResponse =
+        cluster
+            .getSolrClient()
+            .query(
+                COLLECTION,
+                new SolrQuery(
+                    "q",
+                    "{!mlt_content qf=lowerfilt_u^10,lowerfilt1_u^1000 boost=false mintf=0 mindf=0}"
+                        + thirtineenth,
+                    "fq",
+                    "-id:30"));
+    solrDocuments = queryResponse.getResults();
+    expectedIds = new int[] {31, 18, 23, 13, 14, 20, 22, 32, 19, 21};
+    actualIds = new int[solrDocuments.size()];
+    i = 0;
+    for (SolrDocument solrDocument : solrDocuments) {
+      actualIds[i++] = Integer.parseInt(String.valueOf(solrDocument.getFieldValue("id")));
+    }
+
+    Arrays.sort(actualIds);
+    Arrays.sort(expectedIds);
+    System.out.println("DEBUG ACTUAL IDS 1: " + Arrays.toString(actualIds));
+    assertArrayEquals(expectedIds, actualIds);
+
+    queryResponse =
+        cluster
+            .getSolrClient()
+            .query(
+                COLLECTION,
+                new SolrQuery(
+                    "q",
+                    "{!mlt_content qf=lowerfilt_u^10,lowerfilt1_u^1000 boost=true mintf=0 mindf=0}"
+                        + thirtineenth,
+                    "fq",
+                    "-id:30"));
+    solrDocuments = queryResponse.getResults();
+    expectedIds = new int[] {29, 31, 32, 18, 23, 13, 14, 20, 22, 19};
+    actualIds = new int[solrDocuments.size()];
+    i = 0;
+    for (SolrDocument solrDocument : solrDocuments) {
+      actualIds[i++] = Integer.parseInt(String.valueOf(solrDocument.getFieldValue("id")));
+    }
+
+    Arrays.sort(actualIds);
+    Arrays.sort(expectedIds);
+    System.out.println("DEBUG ACTUAL IDS 2: " + Arrays.toString(actualIds));
+    assertArrayEquals(
+        Arrays.toString(expectedIds) + " " + Arrays.toString(actualIds), expectedIds, actualIds);
```

Comment:
not sure that `Arrays.toString(expectedIds) + " " + Arrays.toString(actualIds),` is needed? I think assertArrayEquals prints the differences.